pants_jupyter_plugin/env.py

```diff
@@ -19,3 +22,36 @@ def create(**env_vars: Any) -> Mapping[str, str]:
         else:
             env.pop(name, None)
     return env
+
+
+@dataclass
+class EnvManager:
+    mounted: List[Path] = field(default_factory=list, hash=False)
+
+    def unmount(self) -> Iterator[Path]:
+        """Scrubs sys.path and sys.modules of any contents from previously mounted environments.
+
+        WARNING: This will irreversibly mutate sys.path and sys.modules each time it's called.
+        """
+        while self.mounted:
+            sys_path_entry = self.mounted.pop()
+            sys.path[:] = [
+                entry
+                for entry in sys.path
+                if entry and os.path.exists(entry) and not sys_path_entry.samefile(entry)
+            ]
+
+            for name, module in list(sys.modules.items()):
+                module_path = getattr(module, "__file__", None)
+                if module_path is not None:
+                    if sys_path_entry in Path(module_path).parents:
+                        del sys.modules[name]
+
+            yield sys_path_entry
+
+    def mount(self, path_parts: Iterable[Path]) -> Iterator[Path]:
+        """Mounts an iterable of path parts to sys.path."""
+        for path_entry in path_parts:
+            sys.path.append(str(path_entry))
+            self.mounted.append(path_entry)
```

Comment:
This was broken before, but if it's not been encountered as a problem then clearly it's not a problem in practice yet, but this should probably use `site.addsitedir` - right now `.pth` are not honored and sometimes those are important, including handling old-style namespace packages.